src/Internal/Payment/State/ProcessedState.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Class ProcessedState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WCPay\Exceptions\Order_Not_Found_Exception;
+use WCPay\Internal\Payment\Exception\StateTransitionException;
+use WCPay\Internal\Service\OrderService;
+use WCPay\Vendor\League\Container\Exception\ContainerException;
+
+/**
+ * This state is used when payment is completed on the server, and we need to update date on the plugin side.
+ */
+class ProcessedState extends AbstractPaymentState {
```

Comment:
Based on https://github.com/Automattic/woocommerce-payments/pull/7471#discussion_r1361620696: 


```suggestion
class CapturedState extends AbstractPaymentState {
```

Reply:
Sorry for mistakenly giving my 👍 here, but this is not right.

Because of the patch, I am still to share the state graph so we can discuss it, so let's undo this change for now. `CapturedState` should come after `ProcessedState`, it is not the same.

It would be:
1. `InitialState`
2. `PendingAuthenticationState`
3. `ProcessedState`
4. `CapturedState` or `PendingCaptureState`

The point is not to end up in a `CompletedState`, as a payment is never "complete".

Let's undo this change. We'll finalize state names in a different issue/PR.

Reply:
I have reverted a commit that changed states name.